addons/web/static/src/webclient/company_service.js

```diff
@@ -47,11 +47,20 @@ export const companyService = {
         };
         const allowedCompanyIds = computeAllowedCompanyIds(cids);
         const nextAvailableCompanies = allowedCompanyIds.slice();  // not using a Set because order is important
-        nextAvailableCompanies.add = (companyId) => {
+        nextAvailableCompanies.add = (companyId, putAtStart = false) => {
             if (!nextAvailableCompanies.includes(companyId)) {
-                nextAvailableCompanies.push(companyId);
-                availableCompanies[companyId].child_ids.map(nextAvailableCompanies.add);
+                if (putAtStart) {
+                    nextAvailableCompanies.unshift(companyId);
+                } else {
+                    nextAvailableCompanies.push(companyId);
+                }
+            } else if (putAtStart) {
+                const index = nextAvailableCompanies.findIndex(c => c === companyId);
+                const temp = nextAvailableCompanies[0];
+                nextAvailableCompanies[0] = companyId;
+                nextAvailableCompanies[index] = temp;
```

Comment:
why is the order important?

Reply:
It's just that the first company is the current one, the rest of the order is irrelevant, maybe using something like `asCurrentCompany` would make even more sense instead of `putAtStart` or `unshift` ?

Reply:
Ok, no `unshift` is fine if we keep the internal representation as is (i.e. an array of companies, the first one being the current company). I was just wondering why it matters.